src/submit.py

```diff
@@ -43,6 +47,21 @@ async def _check_git_repo(repo_dir: pathlib.Path):
         click.echo(f"Uncommitted changes in {repo_dir}:")
         click.echo(output)
         click.confirm("Are you sure you want to continue?", abort=True)
+        
+    settings_info = settings.get_settings()
+    has_internet_access = bool(settings_info.permissions)
+    
+    if has_internet_access:
```

Comment:
```suggestion
    if "full_internet" in settings.get_settings()["permissions"]:
```

Reply:
Changed this.
(I don't like it. It's different from how internet is implied in `main.py`. The correct thing imo would be to make the different types of internet permissions clear in `TaskInfo` but that's another repo)